tools/disassemble_sequence.py

```diff
@@ -0,0 +1,792 @@
+            
+        if ".define" in line and current_mode != "":
+            sym, index = re.match(r"\.define ([A-Z_0-9]+)\s+(\d+)", line).groups()
+            if current_mode == "instrument":
+                inst_syms[int(index)] = sym
+            elif current_mode == "drum":
+                drum_syms[int(index)] = sym
+            elif current_mode == "effect":
+                effect_syms[int(index)] = sym
+
+    return (inst_syms, drum_syms, effect_syms)
+
+def get_sequence_symbols(seq_inc):
+    pitches = {}
+    fonts = {}
+
+    for line in seq_inc.readlines():
+        if "PITCH" in line:
+            sym, id = re.match(r"\.define ([A-Z_0-9]+)\s+(\d+)", line).groups()
+            pitches[int(id)] = sym
+        if "SOUNDFONT" in line:
+            sym, id = re.match(r"\.define ([A-Z_0-9]+)\s+(\d+)", line).groups()
+            fonts[int(id)] = sym
+
+    return (pitches, fonts)
+
+control_flow_commands = {
+    0xFF: ['end'],
+    0xFE: ['delay1'],
+    0xFD: ['delay', 'var'],
+    0xFC: ['call', 'addr'],
+    0xFB: ['jump', 'addr'],
+    0xFA: ['beqz', 'addr'],
+    0xF9: ['bltz', 'addr'],
+    0xF8: ['loop', 'u8'],
+    0xF7: ['loopend'],
+    0xF6: ['break'],
+    0xF5: ['bgez', 'addr'],
+    0xF4: ['rjump', 'reladdr8'],
+    0xF3: ['rbeqz', 'reladdr8'],
+    0xF2: ['rbltz', 'reladdr8'],
+}
+
+commands = {}
+commands['seq'] = join(control_flow_commands, {
+    # non-arg commands
+    0xF1: ['allocnotelist', 'u8'],
+    0xF0: ['freenotelist'],
+    0xDF: ['transpose', 's8'],
+    0xDE: ['rtranspose', 's8'],
+    0xDD: ['tempo', 'u8'],
+    0xDC: ['tempochg', 's8'],
+    0xDB: ['vol', 'u8'],
+    0xDA: ['volmode', 's8'],
+    0xD7: ['initchan', 'hex16'],
+    0xD6: ['freechan', 'hex16'],
+    0xD5: ['mutescale', 's8'],
+    0xD4: ['mute'],
+    0xD3: ['mutebhv', 'hex8'],
+    0xD2: ['ldshortvelarr', 'addr'],
+    0xD1: ['ldshortgatearr', 'addr'],
+    0xD0: ['notealloc', 'u8'],
+    0xCE: ['rand', 'u8'],
+    0xCD: ['dyncall', 'addr'],
+    0xCC: ['ldi', 'u8'],
+    0xC9: ['and', 'u8'],
+    0xC8: ['sub', 'u8'],
+    0xC7: ['stseq', 'u8', 'addr'],
+    0xC6: ['stop'],
+    0xC5: ['scriptctr', 'u16'],
+    0xC4: ['runseq', 'u8', 'u8'],
+    # arg commands
+    0x00: ['testchan', 'bits:4'],
+    0x40: ['stopchan', 'bits:4'],
+    0x50: ['subio', 'bits:4'],
+    0x60: ['ldres', 'bits:4', 'u8', 'u8'],
+    0x70: ['stio', 'bits:4'],
+    0x80: ['ldio', 'bits:4'],
+    0x90: ['ldchan', 'bits:4', 'addr'],
+    0xA0: ['rldchan', 'bits:4', 'reladdr16'],
+    0xB0: ['ldseq', 'bits:4', 'u8', 'addr'],
+})
+
+commands['chan'] = join(control_flow_commands, {
+    # non-arg commands
+    0xF1: ['allocnotelist', 'u8'],
+    0xF0: ['freenotelist'],
+    0xEE: ['bendfine', 's8'],
+    0xED: ['gain', 'u8'],
+    0xEB: ['fontinstr'],
+    0xEA: ['stop'],
+    0xE9: ['notepri', 'u8'],
+    0xE8: ['params', 'u8', 'u8', 'u8', 's8', 's8', 'u8', 'u8', 'u8'],
+    0xE7: ['ldparams', 'addr'],
+    0xE6: ['samplbook', 'addr'],
+    0xE5: ['rvrbidx', 'u8'],
```

Comment:
I feel this is much harder to type and understand than simply `reverbindex`

Reply:
This came up in a review of MM cutscene docs, but I wonder if the name `reverbIndex` could be improved upon in general (doesn't have to be addressed in this PR, but it will impact this name down the line):

https://github.com/zeldaret/mm/pull/1164#discussion_r1070473727 (actual cutscene stuff isn't relevant, just the comments on naming `reverbIndex`).

To describe reverb index, there are reverb settings that are provided with audio specifications (i.e. a set of data for each audio session, roughly aligning with 1 session/scene, just like PlayState).

https://github.com/zeldaret/oot/blob/9f09505d34619883748a7dab05071883281c14fd/src/code/audio_init_params.c#L30-L31
(noting reverb index 0 always uses `DEFAULT_REVERB_SETTINGS` by design choice of the devs)

Which set the following settings (named based of audio WIP docs):
```c
typedef struct {
    /* 0x00 */ u8 downsampleRate;
    /* 0x02 */ u16 windowSize;
    /* 0x04 */ u16 decayRatio; // determines how much reverb persists
    /* 0x06 */ u16 subDelay;
    /* 0x08 */ u16 subVolume;
    /* 0x0A */ u16 volume;
    /* 0x0C */ u16 leakRtl;
    /* 0x0E */ u16 leakLtr;
    /* 0x10 */ s8 mixReverbIndex;
    /* 0x12 */ u16 mixReverbStrength;
    /* 0x14 */ s16 lowPassFilterCutoffLeft;
    /* 0x16 */ s16 lowPassFilterCutoffRight;
} ReverbSettings; // size = 0x18
```

However, there are usually 1-2 different reverb settings you can assign to each audio seqPlayer/channel. The reverb index is the index of these predefined/preset reverb settings that you can choose from. Bgm/Fanfares are usually reverbIndex = 0, and all sfx channels are set to reverbIndex = 1:
https://github.com/MNGoldenEagle/oot/blob/fb5bcb9e5e9bfa5e9735c9a3015aa32d896a1767/assets/sequences/000_Sound_Effects.prg.seq#L77

With the exception of the ocarina sfx, which also associates with reverbIndex = 0:
https://github.com/MNGoldenEagle/oot/blob/fb5bcb9e5e9bfa5e9735c9a3015aa32d896a1767/assets/sequences/000_Sound_Effects.prg.seq#L445

Usually when a scene has more reverb, it's only reverbIndex 1 that's given more reverb (ex lost woods), so that ocarina/bgm/fanfares don't get bloated with reverb (keeps reverbIndex = 0 i.e. `DEFAULT_REVERB_SETTINGS`

In audio code, perhaps alternatives could be `reverbPresets`, `reverbPresetId`, `reverbPresetIndex`, `reverbSettingsId`, `reverbSettingsIndex`, or `reverbId`.

Reply:
TBD, but agreed `reverbindex` or similar would be better.